### ISSUE TITLE
NCBI Export in state 'deleted' prevents any other uploads from progressing

### ISSUE THREAD

**Issue by apetkau** [phac-nml/irida#840]
## Describe the bug
If some NCBI Export submission gets a state of `deleted` returned from NCBI then the state for this export will not be updated and other NCBI export jobs will be prevented from progressing.

Specifically, this is the exception:

```
27 Oct 2020 10:06:25,696 ERROR ca.corefacility.bioinformatics.irida.service.export.ExportUploadService:227 - Couldn't connect to FTP site
java.lang.NullPointerException: state for string "deleted" does not exist
        at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:897)
        at ca.corefacility.bioinformatics.irida.model.enums.ExportUploadState.fromString(ExportUploadState.java:114)
        at ca.corefacility.bioinformatics.irida.service.export.ExportUploadService.updateSubmissionForXml(ExportUploadService.java:466)
        at ca.corefacility.bioinformatics.irida.service.export.ExportUploadService.updateRunningUploads(ExportUploadService.java:200)
        at ca.corefacility.bioinformatics.irida.config.services.scheduled.NcbiUploadScheduledTaskConfig.ncbiUploadStatus(NcbiUploadScheduledTaskConfig.java:38)
...
```

## Steps to reproduce the problem
1. Submit many NCBI Export jobs. One of them needs to be switched to state `deleted` by NCBI.
2. The state `deleted` is not registered as a valid state in IRIDA, and will not trigger an error to clean up the export.
3. IRIDA will keep on checking the status of the `deleted` export and failing without making any progress with newer exports.

## Expected behaviour
I expected IRIDA to recognize the `deleted` state and switch the NCBI export to an error state. I also expected IRIDA to make progress on later NCBI exports if earlier ones fail.

## Additional context

The existing mapping for states is found here:

https://github.com/phac-nml/irida/blob/dd3e1a3c65b974c552bf94919c5ae59a23723f5c/src/main/java/ca/corefacility/bioinformatics/irida/model/enums/ExportUploadState.java#L80

Note that the value `DELETED` is mapped to string `"Submission deleted"` in the enum, and not the string `"deleted"` as returned by NCBI.

Additionally, the exception being thrown comes from here:

https://github.com/phac-nml/irida/blob/dd3e1a3c65b974c552bf94919c5ae59a23723f5c/src/main/java/ca/corefacility/bioinformatics/irida/service/export/ExportUploadService.java#L226

Note that this does not switch the existing NCBI export submission to an error state and it also breaks out of the loop for checking later submissions (thus preventing any progress from being made for later submissions).

**Comment by apetkau** [comment]
Just to clarify, the `deleted` state looks like it's coming as a response from the FTP client that the entire submission directory on NCBI has been deleted. It's not coming from the XML file on NCBI listing the status of the submission.

**Comment by ericenns** [comment]
Fixed in #1311
